### PR TITLE
Return HTTP 400 when Accept-Datetime is not RFC1123 + tests

### PR DESCRIPTION
**Pull request by machawk1**

### CODE REVIEW

**Comment by machawk1** [comment]
This would close #579

File: ipwb/util.py [hunk 1]
```diff
@@ -146,6 +146,14 @@ def iso8601ToDigits14(iso8601DateString):
     return d.strftime('%Y%m%d%H%M%S')
 
 
+def isRFC1123Compliant(dtstr):
```
Comment (ibnesayeed) [inline ipwb/util.py]: @machawk1 I think this method is wasteful, duplicates the code, and causes the same logic to be executed twice. This same effect could have been achieved by catching the `ValueError` exception in the `replay.py` file when calling `ipwbUtils.rfc1123ToDigits14(adt)` function.

File: ipwb/replay.py [hunk 2]
```diff
@@ -315,6 +315,9 @@ def queryTimeGate(urir):
     if adt is None:
         adt = ipwbUtils.getRFC1123OfNow()
 
+    if not ipwbUtils.isRFC1123Compliant(adt):
+        abort(400)
+
```
Comment (ibnesayeed) [inline ipwb/replay.py]: No need to introduce `isRFC1123Compliant` function. We can wrap the following line in a `try` block to catch `ValueError` exception instead.

File: ipwb/util.py (hunk 1 above)
Comment (machawk1) [reply ipwb/util.py]: @ibnesayeed I anticipate this function to be useful elsewhere, hence the inclusion in util and not hard-coded to the use case in replay.

File: ipwb/replay.py (hunk 2 above)
Comment (machawk1) [reply ipwb/replay.py]: See response to other comment.

File: ipwb/util.py (hunk 1 above)
Comment (ibnesayeed) [reply ipwb/util.py]: To deal with parsing errors like this, exceptions are generally a better way, no matter how often you use them.

**Pull request merged**